File: clojure/multiples-of-3-or-5/solution.clj
;; https://www.codewars.com/kata/514b92a657cdc65150000006

(ns multiples)

(defn divisible? [a b] (= 0 (mod a b)))

(defn solution [number]
  (reduce +
    (filter
      (fn [n] (or (divisible? n 3) (divisible? n 5)))
      (range 1 number))))

File: rust/multiples-of-3-or-5/solution.rs
// https://www.codewars.com/kata/514b92a657cdc65150000006

const divisors: [i32; 2] = [3, 5];

fn solution(limit: i32) -> i32 {
  let mut sum = 0;

  for i in 3..limit {
    if divisors.iter().any(|div| i % div == 0) {
      sum += i;
    }
  }

  sum
}

